### ISSUE TITLE
번역관련 좀 더 효율적으로 협업하기

### ISSUE THREAD

**Issue by alanleedev** [alanleedev/KoreaSecurityApps#7]
먼저 @DaeHyeoNi @MerHS @sh-cho @ilsubyeega @KENNYSOFT @kunggom 님을 비롯하여 여러 PR로 참여해주신 분들께 감사드립니다.
일단 한국 보안환경이나 한국 내 소프트웨어 개발관련 좋은 사례로 남을 것 같아서 번역을 시작했는데 생각보다 시간이 오래 걸려서 좀 부족한 상태에서 공개하고 PR을 받기 시작했습니다. 
하지만 제가 처음 예상했던 것보다 고쳐야 할 부분이 훨씬 더 많네요. 
일단 저도 이런 식으로 번역을 GitHub에 올려서 하는 것은 처음이고 일단 추가로 공개될 글들이 더 있는 상태인데
전혀 사전 협의 없이 임의로 진행이 되니 약간 비효율 적인 것도 같네요.
일단 여기에 관심 가져주신 분들이 계시니 좀 더 효율적으로 협업을 할 수 있는 방법에 대한 의견을 구합니다.

**Comment by sh-cho** [comment]
Google Docs로 하는게 쉽지 않을까 합니다. 한 페이지를 왼쪽/오른쪽 나눠서 원문/번역 다같이 작성하면 빠르게 할 수 있을 것 같고.. 의견 조율이 필요한 부분이 있으면 닥스 내에 댓글 시스템을 쓰면 될 것 같습니다

**Comment by alanleedev** [comment]
@sh-cho 의견 감사합니다. 일단 협업할 때는 구글 문서가 좀 더 편할 것 같긴하네요. 

일단 다음에 공개 예정인 글을 좀 더 일찍 (비공개로) 번역할 수 있도록 일단 원저라로 부터 글을 공유 받았습니다. 혹시 다른 사람과 공유해서 같이 번역해도 되는지 확인중에 있는데요. 허락이 나면 번역에 참여하실 분들과 공유해서 작업하는 것도 괜찮겠네요. 다음 글 번역에 참여하고 싶으신 분의 여기에 댓글 남겨주세요.

그리고 글을 주제별로 세분화 한 다음에 이슈를 만들어서 혹시 초벌 번역 검토 및 수정하시는 분은 이슈를 본인에게 할당하고 진행하는 방법도 생각해 봤습니다. 그리고 PR도 좀 작게 만들어야지 검토 하기도 수월할 것 같네요. 현재는 누가 글의 어떤 부분을 검토 및 수정하고 있는지 알 수가 없어서 이렇게 하는 것도 필요할 것 같네요.

**Comment by sh-cho** [comment]
참여합니다~

**Comment by kunggom** [comment]
여러 사람이 하다 보니 용어 통일 문제도 있는 것 같습니다.

예를 들어, 제가 올린 PR을 보면 본문에서 `extension`의 번역어로 사용된 `확장기능`을 `확장 프로그램`으로 바꾸었습니다. 이게 구글 크롬 웹스토어에서 사용하는 한국어 번역어였기 때문입니다. 하지만 다른 분들은 여기에 동의하지 않을 수도 있지 않나 생각되네요.

하여튼 여러 사람이 하다 보니 번역어 선택이 갈리는 경우에 대한 대책도 필요하지 않나 생각합니다.

**Comment by alanleedev** [comment]
네. 제대로 협업하려면 용어 사전 같은 것도 필요하죠. 안타깝게도 원저자께서 다음 글은 저에게만 미리 공개하겠다고 하시네요. 아무래도 여러 사람 작업하다보면 사전에 미리 정보가 셀 수 있어서 좀 조심스러운 것 같습니다. 일단 제가 초벌 번역을 하고 다음 번에는 제목 별로 나누어 이슈 만들어서 이슈 가져간 사람이 일차적으로 검토 및 수정 하도록 해야 불필요한 중복을 방지할 것 같네요.

**Comment by alanleedev** [comment]
일단 번역하는 것을 좀 더 효율적으로 하기 위한 가이드를 작성했습니다; https://github.com/alanleedev/KoreaSecurityApps#번역에-참여하는-분을-위한-정보

전 일단 다음 글 미리 받은 것 초벌 번역 시작하려고 합니다. 일단 첫 글 보다는 약간 더 짧고 시간 여유가 더 있을 것 같아서 첫 글 보다는 좀 더 번역 품질을 끌어올려서 올릴 수 있기를 희망합니다. 
일단 원저자 글이 공개 되면 바로 여기에 올리고 주제별로 issue를 만들도록 하겠습니다.
그럼 1차 검토자는 issue를 본인 에게 assign 하시고 2차 검토자에게 PR approver 로 등록 후 서로 필요한 부분 협의 하셔서 최종 승인하면 될 것 같네요. 필요에 따라 다른 사람이 토론에 참여할 수 있지만 일단 2참 검토자에게 approval 권한을 드리면 좋을 것 같군요.

그리고 번역시 용어 협의가 필요하면 https://github.com/alanleedev/KoreaSecurityApps/issues/8 여기에 의견 주세요.

혹시 위 절차에 대해서 추가 의견 있으면 알려주세요.

**Comment by alanleedev** [comment]
@sh-cho @kunggom 
새로운 글에 대한 초벌 번역 올렸습니다. 등록된 이슈 참고해서 작업 진행 부탁드려요.

**Comment by sh-cho** [comment]
@alanleedev
Assignee 수정이 안돼서 Settings - collaborators 에서 추가해주실 수 있나요?

**Comment by alanleedev** [comment]
@sh-cho 추가했습니다. collaborator 아니어도 수정 가능한 방법이 있는지 찾아봐야 겠군요.

**Comment by KENNYSOFT** [comment]
collaborator 아니어도 가능하려면 repo owner를 org로 만들고 team member로 추가해야 합니다.
org를 만들고 repo를 이관하거나 개인 계정을 org로 전환(!)하는 방법이 있습니다.

**Comment by alanleedev** [comment]
@KENNYSOFT 어쨌든 추가하는 단계는 피할 수가 없어 보이네요. 일단 전에 PR 만드신 분은 다 초대했구요. 코멘트 남기시는 분도 collaborator로 추가하도록 할께요.

**Comment by alanleedev** [comment]
@KENNYSOFT 혹시 Organization을 만들고 이관할 필요가 있을까요? 만약 그렇게 되면 리포의 URL도 바뀌는게 맞나요?

**Comment by KENNYSOFT** [comment]
저는 collaborator 추가나 team member 추가나 비슷하다고 생각해서 굳이 필요 없다고 보기는 합니다.
만약 한다면 repo의 URL은 바뀌지만, 이전 주소에서 redirect는 계속 유지됩니다.

**Comment by alanleedev** [comment]
@sh-cho @kunggom 다음 글을 공유받았습니다. 
이번 것은 보안 취약점 공개는 아니라서 공동 번역 작업해도 된다고 허락 받았습니다. 
근데 내용이 이 전에 비하면 짧네요.
일단 전에 의견 받은대로 구글 문서에 글을 올렸구요. 
참여 원하시는 분은 여기 링크에 지메일 주소 남겨주세요: https://forms.gle/X8uDL7gsahQ5LagM8

**Comment by alanleedev** [comment]
참고로 세번째 글도 올라왔습니다.
다음 달 마지막 글 하나만 남았네요.
혹시 여기 원저자와의 영문 인터뷰 기사도 있는데 이것도 번역을 할까요? https://thereadable.co/outsiders-eye-real-questions-to-ask-to-protect-south-koreans-online/

**Comment by sh-cho** [comment]
네 좋습니다

**Comment by alanleedev** [comment]
인터뷰 기사도 번역 허락을 받았구요. 이것은 일단 구글 문서로 작업할께요. 기존에 작업에 참여 신청하신 두 분은 추가했습니다.
 ~~참여하실 분은 여기 이메일 남겨주세요:  https://forms.gle/X8uDL7gsahQ5LagM8~~
여기 글을 올려서 앞으로 필요한 수정은 PR로 합니다.

**Comment by alanleedev** [comment]
참고로 원래 계획에 없었던 글이 추가로 공개되었고 번역글도 올렸습니다.

**Comment by alanleedev** [comment]
번역과 수정에 참여해주시는 분들께 감사합니다.
개인적으로 요즘 많이 바빠져서 저도 잘 신경 못쓰고 있는데요, 다음주에 마지막 글 하나만 공개를 남겨두고 있네요.
특별한 계획없이 시작하였고 많은 분들이 동참해주셨습니다.
저희가 한 내용을 회고(retrospective)하고 기록으로 남길 필요가 있을까요?

**Comment by sh-cho** [comment]
네 좋습니다 :)

**Comment by alanleedev** [comment]
마지막 글 올라왔습니다. 이번에 글이 길고 시간이 없어서 초벌 번역 잘 했나 모르겠네요. 검토를 많이 못했습니다.
